### PR TITLE
boards: pico_spe: add support

### PR DESCRIPTION
**Pull request by KwsBaer**
Add variant of rp2040 with SinglePairEthernet

https://builds.zephyrproject.io/zephyr/pr/81003/docs/boards/index.html#name=spe
https://builds.zephyrproject.io/zephyr/pr/81003/docs/boards/kws/pico_spe/doc/index.html

### CODE REVIEW

File: boards/kws/pico_spe/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2024 KWS Computersysteme Gmbh
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_PICO_SPE
+
+config RP2_FLASH_W25Q080
+	default y
+
```
Comment (nordicjm) [inline boards/kws/pico_spe/Kconfig.defconfig:8]:
```suggestion
```

add `Kconfig` file with:
```
config BOARD_PICO_SPE
	default RP2_FLASH_W25Q080
```

File: boards/kws/pico_spe/doc/img/pico_spe.png
Comment (nordicjm) [inline boards/kws/pico_spe/doc/img/pico_spe.png:1]: Can you replace this with an actual picture of the board rather than a render? Also in webp format, and put it through https://tinypng.com/

File: boards/kws/pico_spe/doc/index.rst [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+     :align: center
+     :alt: Pico-SPE
+
+     Pico-SPE(above)
+     (Images courtesy of KWS Computersysteme Gmbh)
+
+Supported Features
+==================
+
+The pico_spe board configuration supports the following
```
Comment (nordicjm) [inline boards/kws/pico_spe/doc/index.rst:44]:
```suggestion
The ``pico_spe`` board target supports the following
```

File: boards/kws/pico_spe/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+KWS Computersysteme Gmbh. They are equipped with an RP2040 SoC, an on-board LED,
+a USB connector, an SWD interface. The Pico-SPE Additionally contains an
+Microchip LAN8651 10Base-T1S module. The USB bootloader allows the
+ability to flash without any adapter, in a drag-and-drop manner.
+It is also possible to flash and debug the boards with their SWD interface,
+using an external adapter.
+
+Hardware
+********
+- Dual core Arm Cortex-M0+ processor running up to 133MHz
```
Comment (nordicjm) [inline boards/kws/pico_spe/doc/index.rst:19]: +newline

File: boards/kws/pico_spe/doc/index.rst [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+co-processors that are designed for I/O operations. The PIOs run
+a custom instruction set, generated from a custom assembly language.
+PIO programs are assembled using :command:`pioasm`, a tool provided by Raspberry Pi.
+
+Zephyr does not (currently) assemble PIO programs. Rather, they should be
+manually assembled and embedded in source code. An example of how this is done
+can be found at :zephyr_file:`drivers/serial/uart_rpi_pico_pio.c`.
+
+Sample:  SPI via PIO
+====================
```
Comment (nordicjm) [inline boards/kws/pico_spe/doc/index.rst:137]:
```suggestion
Sample: SPI via PIO
===================
```

File: boards/kws/pico_spe/pico_spe-pinctrl.dtsi [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+	adc_default: adc_default {
+		group1 {
+			pinmux = <ADC_CH0_P26>, <ADC_CH1_P27>, <ADC_CH2_P28>, <ADC_CH3_P29>;
+			input-enable;
+		};
+	};
+
+	clocks_default: clocks_default {
+	};
```
Comment (nordicjm) [inline boards/kws/pico_spe/pico_spe-pinctrl.dtsi:69]:
```suggestion
```

File: boards/kws/pico_spe/pico_spe.dts [hunk 6]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2024 KWS Computersysteme Gmbh
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+
+#include "pico_spe-common.dtsi"
```
Comment (nordicjm) [inline boards/kws/pico_spe/pico_spe.dts:9]: not sure why there is a common file when there is only one board, put contents of common file here and remove common file

File: boards/kws/pico_spe/pico_spe_defconfig [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+CONFIG_BUILD_OUTPUT_UF2=y
+CONFIG_BUILD_OUTPUT_HEX=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_RESET=y
+CONFIG_CLOCK_CONTROL=y
+CONFIG_PINCTRL=y
+
+CONFIG_SPI_PL022=y
+CONFIG_SPI_PL022_DMA=y
+CONFIG_DMA_RPI_PICO=y
```
Comment (nordicjm) [inline boards/kws/pico_spe/pico_spe_defconfig:16]:
```suggestion
```

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by KwsBaer** [comment]
Do I missed some code changes?

File: boards/kws/pico_spe/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,296 @@
...
+PIO programs are assembled using :command:`pioasm`, a tool provided by Raspberry Pi.
+
+Zephyr does not (currently) assemble PIO programs. Rather, they should be
+manually assembled and embedded in source code. An example of how this is done
+can be found at :zephyr_file:`drivers/serial/uart_rpi_pico_pio.c`.
+
+Sample: SPI via PIO
+====================
+
+The :zephyr_file:`samples/sensor/bme280/README.rst` sample includes a
```
Comment (kartben) [inline boards/kws/pico_spe/doc/index.rst:139]:
```suggestion
The :zephyr:code-sample:`bme280` sample includes a
```

File: boards/kws/pico_spe/doc/img/pico_spe.png
Comment (kartben) [inline boards/kws/pico_spe/doc/img/pico_spe.png:1]: can you please convert this file to the webp format -- that'll make its size twice (if not more) smaller
You may use cwebp CLI or https://tinypng.com/

File: boards/kws/pico_spe/board.yml [hunk 9]
```diff
@@ -0,0 +1,6 @@
+board:
+  name: pico_spe
+  full_name: SinglePairEthernet Pico
```
Comment (kartben) [inline boards/kws/pico_spe/board.yml:3]: it looks like this should be the following, as this is the official/commercial name?

```suggestion
  full_name: Pico-SPE
```

File: boards/kws/pico_spe/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,296 @@
+.. _pico_spe:
+
+Pico-SPE
+#################
```
Comment (kartben) [inline boards/kws/pico_spe/doc/index.rst:4]:
```suggestion
.. :zephyr:board:: pico_spe
```

File: boards/kws/pico_spe/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,296 @@
...
+- 1 Watchdog timer peripheral
+- Microchip LAN8651 10Base-T1S
+
+
+.. figure:: img/pico_spe.png
+     :align: center
+     :alt: Pico-SPE
+
+     Pico-SPE(above)
+     (Images courtesy of KWS Computersysteme Gmbh)
```
Comment (kartben) [inline boards/kws/pico_spe/doc/index.rst:39]:
```suggestion
```

the `zephyr:board:` directive will add the image automatically :)

File: boards/kws/pico_spe/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,296 @@
...
+
+Pico-SPE
+#################
+
+Overview
+********
+
+The Pico-SPE is a small, low-cost, versatile boards from
+KWS Computersysteme Gmbh. They are equipped with an RP2040 SoC, an on-board LED,
+a USB connector, an SWD interface. The Pico-SPE Additionally contains an
```
Comment (kartben) [inline boards/kws/pico_spe/doc/index.rst:11]:
```suggestion
a USB connector, an SWD interface. The Pico-SPE additionally contains an
```

File: boards/kws/pico_spe/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,296 @@
...
+   :goals: build
+
+.. code-block:: bash
+
+  west flash --runner jlink
+
+Using OpenOCD
+-------------
+
+To use CMSIS-DAP, You must configure **udev**.
```
Comment (kartben) [inline boards/kws/pico_spe/doc/index.rst:172]:
```suggestion
To use CMSIS-DAP, you must configure **udev**.
```

**Review by kartben — CHANGES_REQUESTED** [review]

**Comment by KwsBaer** [comment]
Hi @kartben ,
This is my first PR for this project. I've addressed the requested changes, could you please review and approve them?
Thank you.

File: boards/kws/pico_spe/doc/index.rst (hunk 10 above)
Comment (kartben) [reply boards/kws/pico_spe/doc/index.rst]: @KwsBaer  sorry, made a typo there :( It should be:

```
.. zephyr:board:: pico_spe
```

**Review by kartben — DISMISSED** [review]

**Review by nordicjm — CHANGES_REQUESTED** [review]
Commits needs sorting and squashed for the board, should be 1x commit for vendor prefix update, 1x commit for board and 1x commit for test update

**Comment by KwsBaer** [comment]
@nordicjm is my squash and sorting ok?

File: boards/kws/index.rst [hunk 14]
```diff
@@ -0,0 +1,10 @@
+.. _boards-kws:
+
+KWS Computersysteme Gmbh
```
Comment (nordicjm) [inline boards/kws/index.rst:3]: this can be merged into the other commit

File: dts/bindings/vendor-prefixes.txt [hunk 15]
```diff
@@ -356,6 +356,7 @@ kontron	Kontron S&T AG
 kosagi	Sutajio Ko-Usagi PTE Ltd.
 kvaser	Kvaser
 kvg	Kverneland Group
+kws	KWS Computersysteme Gmbh
```
Comment (nordicjm) [inline dts/bindings/vendor-prefixes.txt:359]: this file to be taken out into separate commit, before this commit

File: tests/drivers/adc/adc_api/boards/pico_spe.overlay [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	zephyr,user {
+		io-channels = <&adc 0>, <&adc 1>;
+	};
+};
+
```
Comment (nordicjm) [inline tests/drivers/adc/adc_api/boards/pico_spe.overlay:12]: this to be taken out into a separate commit, after this commit

File: boards/kws/pico_spe/board.cmake [hunk 17]
```diff
@@ -0,0 +1,36 @@
...
+# And any other SWD debug adapter might also be usable with this configuration.
+
+# Set RPI_PICO_DEBUG_ADAPTER to select debug adapter by command-line arguments.
+# e.g.) west build -b pico_spe -- -DRPI_PICO_DEBUG_ADAPTER=raspberrypi-swd
+# The value is treated as a part of an interface file name that
+# the debugger's configuration file.
+# The value must be the 'stem' part of the name of one of the files
+# in the openocd interface configuration file.
+# The setting is store to CMakeCache.txt.
+if ("${RPI_PICO_DEBUG_ADAPTER}" STREQUAL "")
```
Comment (nordicjm) [inline boards/kws/pico_spe/board.cmake:16]:
```suggestion
if("${RPI_PICO_DEBUG_ADAPTER}" STREQUAL "")
```

File: boards/kws/pico_spe/pico_spe-pinctrl.dtsi (hunk 5 above)
Comment (nordicjm) [reply boards/kws/pico_spe/pico_spe-pinctrl.dtsi:69]: Not addressed
